dat.h

```diff
@@ -163,12 +167,12 @@ enum {
 	Nserial= 20,
 };
 
-int	shelf, slot;
-ulong	aoetag;
-uchar	mac[6];
-int	bfd;		// block file descriptor
-int	sfd;		// socket file descriptor
-vlong	size;		// size of vblade
-vlong	offset;
-char	*progname;
-char	serial[Nserial+1];
+EXTERN int	shelf, slot;
+EXTERN ulong	aoetag;
+EXTERN uchar	mac[6];
+EXTERN int	bfd;		// block file descriptor
+EXTERN int	sfd;		// socket file descriptor
+EXTERN vlong	size;		// size of vblade
+EXTERN vlong	offset;
+EXTERN char	*progname;
+EXTERN char	serial[Nserial+1];
```

Comment:
Declaring an variable by the "EXTERN"-Keyword requires them to get defined somewhere else. Does this code run?

Reply:
This file is included from multiple places, and in one place EXTERN is defined as the empty string so the variables are NOT extern there. That is done in aoe.c.

Reply:
Oh, I missed that fact. This generalisation using another definition of EXTERN may not influence that much code because it is a working solution. Probably a comment would help. Would a patch moving from global variables to a local context be accepted?